wagtail/contrib/modeladmin/tests/test_search_handlers.py

```diff
@@ -0,0 +1,94 @@
+from unittest import mock
+
+from django.test import TestCase
+
+from wagtail.tests.modeladmintest.wagtail_hooks import AuthorModelAdmin, BookModelAdmin
+from wagtail.tests.utils import WagtailTestUtils
+
+
+class DjangoORMSearchHandlerTestCase(TestCase, WagtailTestUtils):
+    fixtures = ['modeladmintest_test.json']
+
+    def setUp(self):
+        self.login()
+
+    def get(self, **params):
+        response = self.client.get('/admin/modeladmintest/author/', params)
+        self.assertEqual(response.status_code, 200)
+        return response
+
+    def test_search_form_absent(self):
```

Comment:
Because search handlers are separate components (or units), we should be able to test them in isolation. So forgetting about requests, authentication, URLs, ModelAdmin classes, and just create an instance directly, and test the behaviour matches our expectations. 

Simulating full requests and testing that everything works together like it should (feature testing) is important, but is a difficult/expensive way to test lots of input combinations. It also makes it more difficult to clearly see how the different variations of the component behave and why.

With this in mind, do you think we could make the tests in this module more 'unit test' like, and let the existing tests worry about all the bits fitting together?

Reply:
In addition to the above, for `WagtailBackendSearchHandler`, we can probably mock the underlying search backend, and just ensure the correct arguments are passed to `search()`... The search backends themselves are well-tested enough. We just want to know that the handler uses the search backend correctly.

Reply:
ah I see, yep will do